src/main/proto/uri.proto

```diff
@@ -20,20 +20,94 @@
  */
 syntax = "proto3";
 
-package uprotocol;
+package uprotocol.v1;
 
 import "google/rpc/status.proto";
 import "google/protobuf/any.proto";
 
-option java_package = "org.eclipse.uprotocol";
+option java_package = "org.eclipse.uprotocol.v1";
+option java_outer_classname = "UUriProto";
 option java_multiple_files = true;
 
 
+// Data representation of uProtocol <b>URI</b>.
+// This class will be used to represent the source and sink (destination) parts 
+// of the Packet, for example in a CloudEvent Packet. <br>
+// UUri is used as a method to uniquely identify devices, services, and resources
+// on the  network.<br>
+// Where software is deployed, what the service is called along with a version and
+// the resources in the service. Defining a common URI for the system allows 
+// applications and/or services to publish and discover each other.<br>
+message UUri {
+  UAuthority authority = 1;
+  UEntity entity = 2;
+  UResource resource = 3;
+  
+  // Any additional details needed
+  repeated google.protobuf.Any details = 10;
```

Comment:
why do we need details? Personally I dont like adding placeholders just for the sake of maybe forgetting something